### ISSUE TITLE
[FEATURE REQUEST] Space character encoding options for custom link names

### ISSUE THREAD

**Issue by ChristinWhite** [IsThereAnyDeal/AugmentedSteam#1782]
### Current Behavior (if any)

For custom links the current behavior for encoding spaces in the `[NAME]` keyword is to just drop them altogether.

For example, for [Ori and the Will of the Wisps](https://store.steampowered.com/app/1057090/Ori_and_the_Will_of_the_Wisps/) using `[NAME]` in a custom link will return `oriandthewillofthewisps`

### Desired Behavior

A slightly better option would be to encode spaces with dash characters. A much flexible solution would be to offer an additional form field asking how spaces should be encoded.

Something along the lines of how [Alfred](https://www.alfredapp.com/) handles this issue for custom search engines:

![CleanShot 2023-10-03 at 22 13 27@2x](https://github.com/IsThereAnyDeal/AugmentedSteam/assets/1746405/41b89c3d-7d63-4c09-8985-a4490016a007)

This could return `Ori-and-the-Will-of-the-Wisps`, `Ori%20and%20the%20Will%20of%20the%20Wisps`, `Ori_and_the_Will_of_the_Wisps` with minimal extra complexity.

### Motivation / Use Case for Changing the Behavior

Obviously, using `[NAME]` in custom links is always going to have issues mapping to other sites that don't use Steam IDs with name variations being common but they could be a little less brittle if we had the option to specify the way spaces in names should be encoded. 

I can't come up with a single site that treats spaces that way. The most common behavior in the gaming sites I've looked at ([Deku Deals](https://www.dekudeals.com/items/ori-and-the-will-of-the-wisps), [CLOUDBASE.GG](https://cloudbase.gg/g/ori-and-the-will-of-the-wisps/), [Xbox Store](https://www.xbox.com/en-US/games/ori-will-of-the-wisps), etc.) all encode spaces as `-` characters but `+` and `%20` are also possibilities, especially with search engines. Steam itself uses `_` characters in their URLs.

It's not a perfect solution, this will still fail for different editions of games, different letter cases, etc. but it's conceptually a fairly simple improvement with a decent payoff.

It would also be nice if we could rearrange 3rd-party site links but I'll leave that for another day. Thanks for maintaining such an awesome plugin, Steam's site is so much more powerful now thanks to your work!

**Comment by candela97** [comment]
We use the app name as shown on the page, remove some special characters like trademark symbols, then encode it with `encodeURIComponent` (so whitespaces are encoded as `%20`), so there must be something else going on here. What browser are you using, and can you show a screenshot of your custom link from our Options page?

**Comment by ChristinWhite** [comment]
That's interesting and doing some more testing today I think it is behaving the way you suggested and mistakes in my testing last night. I am using [Arc](https://arc.net/) and here are the settings I am using:

![CleanShot 2023-10-04 at 10 07 23@2x](https://github.com/IsThereAnyDeal/AugmentedSteam/assets/1746405/9d05de4d-a4a7-493b-ace5-25ec4b29b02b)

I mistakenly thought Augmented Steam was dropping the spaces on Deku Deals but looking at the link in Steam, it is using `%20`. When Deku Deals doesn't find a game at the URL it is redirecting the string you're looking for into a search and they are the ones dropping the `%20` in the search.

For CLOUDBASE.GG, I would have sworn it also was dropping spaces last night but when I tested it again this morning after turning off all other extensions it is also using `%20`. I turned my other extensions back on and it is continuing to behave as expected so I think I may have been completely mistaken.

All of that said, I still think an option for how you encode spaces would be helpful but if you're just doing the standard JS URL encoding it may not be worth trying to rework it.

**Comment by Korb** [comment]
I agree, there is such a problem at the moment. As an example, I will give the site appagg.com. For it, I specified the link

`https://appagg.com/search/[NAME]:steam-games/?hl=en`

However, instead of

`https://appagg.com/search/cyberpunk%202077:all/?hl=en`

page

`https://appagg.com/search/cyberpunk+2077:all/?hl=en`

opens.
